### PR TITLE
Introduce naming of the gophers component

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Moved watch pods and stability-checker to prow
- Renamed `binding-usage-controller` to `service-binding-usage-controller`
- Renamed and moved etcd-backup and etcd-tls-setup from `/tools` to `/components`

**Related issue(s)**

### CODE REVIEW

**Comment by klaudiagrz** [comment]
Please check if there will be any broken links in the documentation as a result of this change.

File: components/service-binding-usage-controller/README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# Binding Usage Controller
```
Comment (jasiu001) [inline components/service-binding-usage-controller/README.md]: What about convention of component's name  in `resources/*/values.yaml` files. For example for this component version could be found in `resources/service-catalog-addons/values.yaml` under the key `binding_usage_controller`. Should the key change?

Comment (polskikiel) [reply components/service-binding-usage-controller/README.md]: discussed, we will change the key and chart name.

File: resources/service-catalog/charts/etcd-stateful/templates/05-backup-job.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ spec:
           serviceAccountName: {{ template "etcd-fullname" . }}-backup
           containers:
           - name: etcd-backup
-            image: "{{ .Values.global.containerRegistry.path }}/{{ .Values.global.etcd_backup.dir }}etcd-backup:{{ .Values.global.etcd_backup.version }}"
+            image: "{{ .Values.global.containerRegistry.path }}/{{ .Values.global.etcd_backup_job.dir }}etcd-backup:{{ .Values.global.etcd_backup.version }}"
```
Comment (jasiu001) [inline resources/service-catalog/charts/etcd-stateful/templates/05-backup-job.yaml]: missed version part, should be `.Values.global.etcd_backup_job.version` is `.Values.global.etcd_backup.version`

Comment (polskikiel) [reply resources/service-catalog/charts/etcd-stateful/templates/05-backup-job.yaml]: good catch, fixed

**Comment by polskikiel** [comment]
I would not change the images names because the task is not defining this and we use those names only internally in our charts. Those names also didn't change that much. 

I have also changed the name `binding-usage-controller` to `service-binding-usage-controller` in the following files: 
- resources/service-catalog-addons/charts/service-binding-usage-controller/Chart.yaml
- resources/service-catalog-addons/requirements.yaml

This files I won't change: 
- resources/service-catalog-addons/charts/service-binding-usage-controller/templates/function-usage-kind.yaml - `buc` name is used here only in the comment
- resources/service-catalog-addons/charts/service-binding-usage-controller/templates/deployment.yaml - `buc` name is used here as a image name.

**Pull request merged**